keps/sig-apps/3850-backoff-limits-per-index-for-indexed-jobs/README.md

```diff
@@ -0,0 +1,1424 @@
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+
+Additionally, for features that are introducing a new API field, unit tests that
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+The change is opt-in, it doesn't impact already running workloads.
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+###### What specific metrics should inform a rollback?
+
+A substantial increase in the `job_sync_duration_seconds`.
+
+Also, a substantial increase in the total number of pods, as it may take
+additional time to get the finalizers removed.
+
+Additionally, a substantial increase in the difference of
+`terminated_pods_tracking_finalizer_total` for the `add` and `delete` labels may
+indicate that it takes too long to delete the finalizers.
+
+The feature is opt-in so in case of issues it is enough not to use the
+backoffLimitPerIndex API field.
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+It will be tested manually prior to beta launch.
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+By the presence of the `.spec.backoffLimitPerIndex` field in the jobs.
```

Comment:
for beta: let's consider a metric: perhaps `job_completed_index` with a label for `status=(failed|succeeded)`

Reply:
I like the idea, the only doubt is if it should be incrementing on regular indexed jobs. Then, it could skew results. Another label might be needed to distinguish.

Reply:
I proposed `job_finished_indexes_total`. I prefer `finished`, because currently in Jobs completed is often used as synonym to succeeded, which could be misleading.

Reply:
+1
I think it's ok to do it for regular indexed jobs.

Reply:
Ok, proposed also the `backoffLimit=(perIndex|global)` to distinguish